datajunction-server/datajunction_server/models/materialization.py

```diff
@@ -114,6 +145,11 @@ class GenericMaterializationConfigInput(BaseModel):
     # Spark config
     spark: Optional[SparkConf]
 
+    # The time window to lookback when overwriting materialized datasets
+    # This will only be used if a time partition was set on the node and
+    # the materialization strategy is INCREMENTAL_TIME
+    lookback_window: Optional[str]
```

Comment:
Good point, I was actually thinking that this would be something like `1 DAY`, which is parseable via Spark SQL's time string parser. I'll set that as the default.

Reply:
I just looked at your example form image and realized one thing about this. I think this makes most sense for the incremental_time, but for all the rest it should probably default to all-data in the source table. I think we should walk through these examples together... or I'll write them up after my next meeting.

Reply:
Ideally the form elements would react to the values of others but for now maybe we could just do some defaults and  describe the assumptions somewhere. Maybe for now we should let the users do whatever they want but eventually we should probably have more guide rails. Example:

Strategy:     
- INCREMENTAL_TIME:
   - LookBackWindow:   Any* (default: 1)                      
   - Schedule:    Any
- FULL:
   - LookBackWindow:   Any* (default: all)      
   - Schedule:    Disabled (No Backfill Needed)
- SNAPSHOT:
   - LookBackWindow:   Any* (default: all)      
   - Schedule:    Any (because it goes into a fixes snapshot space)
- VIEW:
   - LookBackWindow:   Any
   - Schedule:    Any (but probably default to @daily)

Look-back Window of Any* means that user can specify any INT value, but only if a temporal partition exists.

Reply:
Agreed -- the UI should be changed so that if someone selects "incremental time", it'll display a lookback window form field, but if they select any other thing, it won't allow the option for lookback window. 

I also just added some parsing to handle different lookback window values, which are actually just any Spark SQL compatible `INTERVAL` values (someone could specify it to be `1 day` or `5 hours 30 seconds 1 milliseconds 1 microseconds`, for example, and Spark will handle the parsing).